### PR TITLE
chore: fixing pub deps to allow publish

### PR DESCRIPTION
**Pull request by erickzanardo**
# Description

Pub was complaining that we didn't had a few direct deps on our pubspec.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the
relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples`.

## Breaking Change

Does your PR require Flame users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change. (Indicate it in the [Conventional Commit] prefix with a `!`,
  e.g. `feat!:`, `fix!:`).
- [ ] No, this is *not* a breaking change.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database].
Indicate which of these issues are resolved or fixed by this PR, i.e. Fixes #xxxx*

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

File: packages/flame_test/pubspec.yaml [hunk 1]
```diff
@@ -15,6 +15,7 @@ dependencies:
     sdk: flutter
   meta: ^1.7.0
   test: ^1.17.12
+  vector_math: ^2.1.2
```
Comment (spydon) [inline packages/flame_test/pubspec.yaml:18]: Shouldn't this be `any` too?

Comment (erickzanardo) [reply packages/flame_test/pubspec.yaml:18]: we can't have published packages on pub that have a dep with any on its pubspec, so I added the last one

Comment (spydon) [reply packages/flame_test/pubspec.yaml:18]: Alright, seems like it is too new and that we have to use `2.1.1` then?

Comment (erickzanardo) [reply packages/flame_test/pubspec.yaml:18]: yeah, which is weird because it works on my machine, but either way I am downgrading it and upload a commit

**Review by spydon — APPROVED** [review]

**Pull request merged**